capa/features/extractors/dnfile/helpers.py

```diff
@@ -299,22 +316,93 @@ def get_dotnet_unmanaged_imports(pe: dnfile.dnPE) -> Iterator[DnUnmanagedMethod]
         # like kernel32.CreateFileA
         yield DnUnmanagedMethod(token, module, method)
 
+def get_nested_class_table(pe):
+    nested_class_table = {}
+    
+    # Used to find nested classes in typedef
+    for rid, nestedclass in iter_dotnet_table(pe, dnfile.mdtable.NestedClass.number):
+        assert isinstance(nestedclass, dnfile.mdtable.NestedClassRow)
+        nested_class_table[nestedclass.NestedClass.row_index] = nestedclass.EnclosingClass.row_index
+
+    return nested_class_table
 
+def get_typedef_class_table(pe):
+    typedef_class_table = []
+    
+    for rid, typedef in iter_dotnet_table(pe, dnfile.mdtable.TypeDef.number):
+        assert isinstance(typedef, dnfile.mdtable.TypeDefRow)
+        typedef_class_table.append((typedef.TypeName, typedef.TypeNamespace))
+        
+    return typedef_class_table
+    
+def typedef_helper(index, nested_class_table, typedef_class_table, n, name):
+    # Append the current typeref name
+    n.append(name)
+    
+    if nested_class_table[index] in nested_class_table:
+        typedef_helper(nested_class_table[index], nested_class_table, typedef_class_table, n, typedef_class_table[nested_class_table[index]-1][0])
```

Comment:
Can we make this function iterative instead? This avoids having to seed the function and potential stack issues when dealing with complex/obfuscated .NET files.

Reply:
Agreed, implemented a while loop to change function to iterative as well.